pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2553,9 +2570,17 @@ func mergeFilterChains(httpFilterChain, tcpFilterChain []*listener.FilterChain)
 			fc.FilterChainMatch = &listener.FilterChainMatch{}
 		}
 
-		fc.FilterChainMatch.ApplicationProtocols = append(fc.FilterChainMatch.ApplicationProtocols, plaintextHTTPALPNs...)
+		httpMatch := false
+		for _, p := range fc.FilterChainMatch.ApplicationProtocols {
+			if p == plaintextHTTPALPNs[0] {
```

Comment:
nit: This seems like it might not be as reliable as needed, especially if we make future changes. Should we check that it contains all of `plaintextHTTPALPNs`?